concordium-consensus/src/Concordium/GlobalState/Persistent/BlobStore.hs

```diff
@@ -1047,6 +1053,84 @@ instance DirectBlobStorable m a => BlobStorable m (Nullable (EagerBufferedRef a)
 instance (Applicative m, Cacheable m a) => Cacheable m (EagerBufferedRef a) where
     cache (EagerBufferedRef ioref v) = EagerBufferedRef ioref <$> cache v
 
+-- |A reference that is generally not retained in memory once it has been written to disk.
+--
+-- This is essentially a simplified version of 'BufferedRef', where @BRBoth ref v@ is simply
+-- replaced with @URBlobbed ref@.
+data UnbufferedRef a
+    = URBlobbed !(BlobRef a)
+    -- ^A reference that is already on disk
+    | URMemory {urIORef :: !(IORef (Nullable (UnbufferedRef a))), urValue :: !a}
+    -- ^A reference that is in memory and may have been written to disk.
+    -- If the reference has not been written, the 'urIORef' will contain 'Null'.
+    -- If it has been written, it will contain @Some (URBlobbed b)@ for a 'BlobRef' @b@ that
+    -- holds the stored value.
```

Comment:
The need for the recursive type here is unclear and should be made clear.

It seems again that it would be a lot clearer if instead it was just `BlobRef a` instead of `UnbufferedRef a`, so at the very least it needs to be clearly explained why this is not.

Reply:
I've replaced it with a `BlobRef` instead.